Only one role per User can be assigned

giocalitri opened the pull request:
#### What are the relevant tickets?
closes #1034

#### What's this PR do?
Enforces the existence of only one role type per user

#### Where should the reviewer start?
`roles/models.py`

#### How should this be manually tested?
try to assign `Staff` and `Admin` to the same user in the django Admin

#### Any background context you want to provide?
This is a limitation of the library we are using and given that there are thoughts to create a new `reviewer` role, it makes sense to be sure there is not confusion on this front.

There is a known issue with this code: even if in the Admin you get an error, there will still be a confirmation message. AFAIK this cannot be removed because the confirmation message is generated after the error message. I do not think this is a problem given that only people who access the Admin can see it.
![screen shot 2016-09-14 at 4 36 01 pm](https://cloud.githubusercontent.com/assets/1005204/18529010/5efec162-7a99-11e6-9df0-606932ebb960.png)

noisecapella commented:
Code looks fine overall, testing functionality now...

noisecapella commented:
Just to clarify, a user can be 'staff' of many programs, so the thing that this PR prevents is a user being staff of any program and instructor of any program at the same time?

noisecapella commented:
When I add a new 'instructor' role to a user who is staff, I get this error:
![screenshot from 2016-09-15 10-41-47](https://cloud.githubusercontent.com/assets/863262/18554259/9df0bb4e-7b31-11e6-9e0d-1d0d81feaccb.png)

giocalitri commented:
> Just to clarify, a user can be 'staff' of many programs, so the thing that this PR prevents is a user being staff of any program and instructor of any program at the same time?

yes

giocalitri commented:
I see the error as well: I will try to deal with it.

giocalitri commented:
Ok I fixed the problem and in the process I implemented a better solution that prevents also the problem of the previous screenshot from happening
![screen shot 2016-09-15 at 2 38 38 pm](https://cloud.githubusercontent.com/assets/1005204/18562536/1c727cda-7b52-11e6-9a31-ca309512007c.png)

noisecapella commented:
Looks great :+1:

giocalitri commented:
@noisecapella done

noisecapella commented:
:+1:
